src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -908,15 +909,20 @@ private StringBuilder getFeedbackSessionResultsForQuestionInCsvFormat(
             exportBuilder.append(statistics).append(Const.EOL);
         }
 
-        exportBuilder.append(questionDetails.getCsvDetailedResponsesHeader());
-
         List<String> possibleGiversWithoutResponses = fsrBundle.getPossibleGivers(question);
         List<String> possibleRecipientsForGiver = new ArrayList<String>();
         String prevGiver = "";
-
+        if (allResponses.isEmpty()) {
+            exportBuilder.append(questionDetails.getCsvDetailedResponsesHeader(false, 0));
+        }
```

Comment:
Might want to leave a comment explaining this (it appears to set up a default header if no responses are present, if I'm reading this correctly)